libs/coin-modules/coin-polkadot/src/network/node/apiPromise.ts

```diff
@@ -11,10 +12,22 @@ export default async function () {
     const headers = coinConfig.node.credentials
       ? { Authorization: "Basic " + coinConfig.node.credentials }
       : undefined;
-    api = await ApiPromise.create({
-      provider: new HttpProvider(coinConfig.node.url, headers),
-      noInitWarn: true, //to avoid undesired warning (ex: "API/INIT: polkadot/1002000: Not decorating unknown runtime apis")
-    });
+
+    const nodeURL = coinConfig.node.url;
+
+    if (nodeURL.startsWith("ws://") || nodeURL.startsWith("wss://")) {
```

Comment:
Just extract the provider in a separate variable and keep out of the "if scope" the creation of the ApiPromise.